internal/browser/browser.go

```diff
@@ -26,11 +28,36 @@ type Opener interface {
 
 type StderrPrint struct{}
 
+func openURL(url string) error {
+	var err error
+	switch runtime.GOOS {
+	case "darwin": // macOS
+		err = exec.Command("open", url).Start()
+	case "linux":
+		providers := []string{"xdg-open", "x-www-browser", "www-browser"}
+
+		// There are multiple possible providers to open a browser on linux
+		// One of them is xdg-open, another is x-www-browser, then there's www-browser, etc.
+		// Look for one that exists and run it
+		for _, provider := range providers {
+			if _, err := exec.LookPath(provider); err == nil {
+				err = exec.Command(provider, url).Start()
+			}
+		}
+	case "windows":
+		err = exec.Command("rundll32", "url.dll,FileProtocolHandler", url).Start()
+	default:
+		err = fmt.Errorf("unsupported platform")
+	}
+	return err
+}
+
 func (p *StderrPrint) Open(u *url.URL) error {
 	fmt.Fprintf(
-		os.Stderr,
-		"Please open the following URL in your web browser to authenticate:\n\n\t%s\n\n",
+		os.Stdout,
```

Comment:
Why the switch to stdout? It does mean the object should probably be renamed, as `StderrPrint` now no longer only prints to stderr

Originally this was stderr because I figured it followed UNIX best practices (stdout for machine-consumable output, stderr for human-only output)

Reply:
yeah I was wondering about the choice for stderr originally. You make a good point about stdout being used for machine output. Will revert it back to stderr.